### ISSUE TITLE
Investigate adding built-in keyboard resize/reorder shortcuts to DetailsList

### ISSUE THREAD

**Issue by smhigley** [microsoft/fluentui#23041]
This would be in addition to the recommended menu approach in [our example](https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist/keyboardcolumnreorderresize), which will always be the safer and more discoverable option.

That said, we've had several requests to build in a keyboard shortcut to the base DetailsList functionality, and I think we could look in to implementing that, but we'd need to make sure it is discoverable to end users. Some options to do so include:
- an info bubble above the table that shows helper text when column headers are in focus (this would need to be localized by consumers)
- highlight the resize handler hover style when column headers are in focus
- ensure column headers have an accessible description that includes the keyboard shortcuts (also would need to be localized, if separate from the info bubble)
- not `aria-keyshortcuts` alone for support reasons

The specific shortcut possibilities for now are:
- ctrl + left/right arrows for resize
- ctrl + [ and ] (square bracket keys) for reorder
- ctrl X / ctrl V for reorder, though this is probably not a good idea, since it would be easy to do so accidentally while trying to copy text

Reorder is the stickiest one, since it doesn't have a lot of unified precedent. We also need to make sure we don't conflict with any browser or OS shortcuts.
